src/modules/75-cf/components/AddKeyDialog/__tests__/AddKeyDialog.test.tsx

```diff
@@ -6,34 +6,59 @@
  */
 
 import React from 'react'
-import { render, fireEvent } from '@testing-library/react'
+import { render, RenderResult, waitFor, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import * as cfServices from 'services/cf'
 import { TestWrapper } from '@common/utils/testUtils'
-import AddKeyDialog from '../AddKeyDialog'
+import { CF_DEFAULT_PAGE_SIZE, EnvironmentSDKKeyType } from '@cf/utils/CFUtils'
+import AddKeyDialog, { AddKeyDialogProps } from '../AddKeyDialog'
 
-jest.mock('services/cd-ng', () => ({
-  useGetEnvironmentListForProject: jest.fn().mockImplementation(() => ({
-    data: {
+const apiKeysMock = [
+  {
+    apiKey: 'mock',
+    identifier: 'mock identifier',
+    key: 'dummy ',
+    name: 'dummy api key name',
+    type: EnvironmentSDKKeyType.SERVER
+  }
+]
+
+describe('Test AddKeyDialog', () => {
+  const useAddAPIKeyMock = jest.spyOn(cfServices, 'useAddAPIKey')
+  const useGetAllAPIKeysMock = jest.spyOn(cfServices, 'useGetAllAPIKeys')
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    useAddAPIKeyMock.mockReturnValue({
+      cancel: jest.fn(),
+      error: null,
+      loading: false,
+      mutate: jest.fn().mockResolvedValue(null)
+    })
+
+    useGetAllAPIKeysMock.mockReturnValue({
+      loading: false,
+      error: null,
+      refetch: jest.fn(),
+      absolutePath: '',
+      response: {} as Response,
+      cancel: jest.fn(),
       data: {
-        content: [
-          {
-            identifier: 'abc',
-            name: 'abc'
-          }
-        ]
+        itemCount: Math.ceil(1 / CF_DEFAULT_PAGE_SIZE),
```

Comment:
This should just be the whole item count, so you can just do `1` (since you know the size), or to make it more flexible...
```suggestion
        itemCount: apiKeysMock.length
```

Reply:
done